prow/external-plugins/cherrypicker/server_test.go

```diff
@@ -346,152 +347,217 @@ func testCherryPickPR(clients localgit.Clients, t *testing.T) {
 			t.Errorf("Cleaning up client: %v", err)
 		}
 	}()
-	if err := lg.MakeFakeRepo("foo", "bar"); err != nil {
-		t.Fatalf("Making fake repo: %v", err)
-	}
-	if err := lg.AddCommit("foo", "bar", initialFiles); err != nil {
-		t.Fatalf("Adding initial commit: %v", err)
-	}
-	expectedBranches := []string{"release-1.5", "release-1.6", "release-1.8"}
-	for _, branch := range expectedBranches {
-		if err := lg.CheckoutNewBranch("foo", "bar", branch); err != nil {
-			t.Fatalf("Checking out pull branch: %v", err)
-		}
-	}
-	if err := lg.CheckoutNewBranch("foo", "bar", "cherry-pick-2-to-release-1.5"); err != nil {
-		t.Fatalf("Checking out existing PR branch: %v", err)
+
+	botUser := &github.UserData{Login: "ci-robot", Email: "ci-robot@users.noreply.github.com"}
+
+	getSecret := func() []byte {
+		return []byte("sha=abcdefg")
 	}
 
-	ghc := &fghc{
-		orgMembers: []github.TeamMember{
-			{
-				Login: "approver",
-			},
-			{
-				Login: "merge-bot",
-			},
-		},
-		prComments: []github.IssueComment{
-			{
-				User: github.User{
-					Login: "developer",
-				},
-				Body: "a review comment",
-			},
-			{
-				User: github.User{
-					Login: "approver",
-				},
-				Body: "/cherrypick release-1.5\r\n/cherrypick release-1.8",
-			},
-			{
-				User: github.User{
-					Login: "approver",
-				},
-				Body: "/cherrypick release-1.6",
-			},
-			{
-				User: github.User{
-					Login: "fan",
-				},
-				Body: "/cherrypick release-1.7",
-			},
-			{
-				User: github.User{
-					Login: "approver",
-				},
-				Body: "/approve",
+	logger := logrus.StandardLogger().WithField("client", "cherrypicker")
+	doNotPushFn := func(forkName, newBranch string, force bool) error { return nil }
+	testCases := []struct {
+		name         string
+		testOrgName  string
+		testRepoName string
+		newServer    func(githubClient) *Server
+		newGitHubPR  func(string) github.PullRequest
+	}{
+		{
+			name:         "Push to fork",
+			testOrgName:  "foo",
```

Comment:
This is identical in both testcases, can we move it out to de-duplicate? Same for the `newgithubPR`